waiter/test/waiter/token_test.clj

```diff
@@ -836,6 +874,49 @@
           (is (= token-metadata-2 (select-keys token-description sd/token-metadata-keys)))
           (is (= (merge service-description-2 token-metadata-2) token-description)))))))
 
+(deftest test-store-service-description-backwards-compatibility
```

Comment:
Since we can reindex I'm not sure the work done here building this test was necessary and in other places was necessary.  We'll have more code to deal with going forward.

Reply:
This prevents any token updates from failing after we deploy and before we reindex. we can always remove the backwards compatibility code in a subsequent release after reindexing all the clusters.